tools/check-changelog/check-changelog.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-changelog"
+)
+
+var (
+	skipLabelName = "skip-changelog-check"
+	skipTitles    = []string{"chore", "test", "doc", "ci"} // Dependabot uses chore.
+)
+
+func main() {
+	var (
+		title      = os.Getenv("PR_TITLE")
+		number     = os.Getenv("PR_NUMBER")
+		jsonLabels = os.Getenv("PR_LABELS")
+	)
+	if title == "" || number == "" || jsonLabels == "" {
+		log.Fatal("Environment variables PR_TITLE, PR_NUMBER and PR_LABELS are required")
+	}
+	var labels []string
+	if err := json.Unmarshal([]byte(jsonLabels), &labels); err != nil {
+		log.Fatalf("PR_LABELS is not a stringified JSON array: %v", err)
+	}
+
+	filePath := fmt.Sprintf(".changelog/%s.txt", number)
+	content, errFile := os.ReadFile(filePath)
+	if errFile == nil { // Always validate changelog file if present, skip logic is not considered in this case
+		entry := changelog.Entry{
+			Body: string(content),
+		}
+		if err := entry.Validate(); err != nil {
+			log.Fatalf("Error validating changelog file: %s, err: %v", filePath, err)
+		}
+		fmt.Printf("Changelog file is valid: %s\n", filePath)
+		return
+	}
+
+	if skipTitle(title) {
+		fmt.Println("Skipping changelog check because PR title")
+		return
+	}
+
+	if skipLabel(labels) {
+		fmt.Printf("Skipping changelog check because PR label found: %s\n", skipLabelName)
+		return
+	}
+
+	log.Fatalf("Consider using label %s if this PR doesn't need a changelog.\nChangelog file not found: %s, err: %v", skipLabelName, filePath, errFile)
```

Comment:
Would it be worth logging an additional line referencing to the guidelines? https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/changelog-process.md#changes-that-should-have-a-changelog-entry

Reply:
if it's ok I'll add a sentence to suggest reviewing the contribution guide as the link is very long and might change and get outdated